implementation/src/main/java/io/smallrye/config/mapper/ConfigurationInterface.java

```diff
@@ -0,0 +1,1063 @@
+package io.smallrye.config.mapper;
+
+import static io.smallrye.config.mapper.ConfigMappingProvider.skewer;
+import static org.objectweb.asm.Type.getDescriptor;
+import static org.objectweb.asm.Type.getInternalName;
+import static org.objectweb.asm.Type.getType;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.AnnotatedType;
+import java.lang.reflect.Array;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.GenericArrayType;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.WildcardType;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.microprofile.config.spi.Converter;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+import io.smallrye.common.constraint.Assert;
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.WithConverter;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithName;
+import io.smallrye.config.inject.InjectionMessages;
+import sun.misc.Unsafe;
+
+/**
+ * Information about a configuration interface.
+ */
+public final class ConfigurationInterface {
+    static final boolean usefulDebugInfo = true;
```

Comment:
This should be set to the result of some system property like `io.smallrye.config.mapper.useful-debug-info` which defaults to `false` (since there's a perf cost).